python-spec/pyproject.toml

```diff
@@ -7,7 +7,7 @@ name = "somacore"
 description = "Python-language API specification and base utilities for implementation of the SOMA system."
 dynamic = ["version"]
 readme = "./README.md"
-dependencies = ["attrs>=22.1", "pyarrow", "typing-extensions"]
+dependencies = ["attrs>=22.1", "numpy", "pyarrow", "typing-extensions"]
```

Comment:
I think you need `numpy>=1.21` as some of the `numpy.typing` stuff was introduced in that release (eg., `numpy.typing.NDArray`)

Reply:
Added.